### PR TITLE
storage e2e: remove obsolete storage_operation_status_count usage

### PR DESCRIPTION
**Pull request by pohly**
#### What type of PR is this?

/kind cleanup
/kind failing-test

#### What this PR does / why we need it:

That metric was removed in https://github.com/kubernetes/kubernetes/pull/102801
and therefore the tests which checked it started to fail in gce-serial.

#### Which issue(s) this PR fixes:
Fixes #105965

#### Special notes for your reviewer:

I have not removed the `storage_operation_status_count usage` usage in `testsuites/base.go` because it doesn't seem to cause problems and I am not sure how much of the surrounding code there is supposed to be kept.

#### Does this PR introduce a user-facing change?
```release-note
NONE
```

### CODE REVIEW

File: test/e2e/storage/volume_metrics.go [hunk 1]
```diff
@@ -706,27 +702,9 @@ func verifyMetricCount(oldMetrics, newMetrics *storageControllerMetrics, metricN
 		oldLatencyCount = 0
 	}
 
-	oldStatusCount := int64(0)
```
Comment (msau42) [inline test/e2e/storage/volume_metrics.go:709]: I think we still want to keep the counts check. The status count actually got moved to the existing "storage_operation_duration_seconds" metric.

Comment (pohly) [reply test/e2e/storage/volume_metrics.go:709]: @gnufied: can you come up with a better fix?

I lack background about what these metrics do and I don't have time for something more complex than simply removing some apparently out-dated checks right now.

Comment (msau42) [reply test/e2e/storage/volume_metrics.go:709]: @mauriciopoppe has gladly offered to take a look at this.

**Comment by pohly** [comment]
/close

Solved via https://github.com/kubernetes/kubernetes/pull/106609

**Pull request closed without merging**